packages/components/src/components/button-pure/button-pure.tsx

```diff
@@ -123,6 +130,19 @@ export class ButtonPure {
   public onClick(e: MouseEvent): void {
     if (this.isDisabledOrLoading) {
       e.stopPropagation();
+      return;
+    }
+
+    if (this.form && this.internals.form) {
+      e.preventDefault();
+      switch (this.type) {
+        case 'submit':
+          this.internals.form.requestSubmit();
```

Comment:
Will this also send the button value? `requestSubmit` also takes in a `submitter` parameter where the button can be passed. It this necessary? Also add an E2E testing if the button value is included in the form.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/requestSubmit#submitter